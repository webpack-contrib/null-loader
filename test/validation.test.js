import webpack from './helpers/compiler';

describe('validation', () => {
  it('without options', async () => {
    const config = {
      loader: {
        test: /\.js$/,
      },
    };

    const stats = await webpack('fixture.js', config);
    const { warnings, errors } = stats.toJson();

    expect(warnings).toMatchSnapshot('warnings');
    expect(errors).toMatchSnapshot('errors');
  });

  it('with empty options', async () => {
    const config = {
      loader: {
        test: /\.js$/,
        options: {},
      },
    };

    const stats = await webpack('fixture.js', config);
    const { warnings, errors } = stats.toJson();

    expect(warnings).toMatchSnapshot('warnings');
    expect(errors).toMatchSnapshot('errors');
  });

  it.only('with unknown options', async () => {
    const config = {
      loader: {
        test: /\.js$/,
        options: { unknown: 'foo-bar' },
      },
    };

    const stats = await webpack('fixture.js', config);
    const { warnings, errors } = stats.toJson();

    expect(warnings).toMatchSnapshot('warnings');
    expect(errors).toHaveLength(1);
  });
});
