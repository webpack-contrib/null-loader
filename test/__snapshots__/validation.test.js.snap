// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation with empty options: errors 1`] = `Array []`;

exports[`validation with empty options: warnings 1`] = `Array []`;

exports[`validation with unknown options: warnings 1`] = `Array []`;

exports[`validation without options: errors 1`] = `Array []`;

exports[`validation without options: warnings 1`] = `Array []`;
